src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/CrossCaseTypeSearchRequest.java

```diff
@@ -45,12 +56,75 @@ public JsonNode getSearchRequestJsonNode() {
     }
 
     public boolean isMultiCaseTypeSearch() {
-        return caseTypeIds.size() > 1;
+        return multiCaseTypeSearch;
     }
 
     private void validateJsonSearchRequest() {
         if (!searchRequestJsonNode.has(QUERY)) {
             throw new BadSearchRequest("missing required field 'query'");
         }
     }
+
+    public List<String> getSourceFilterAliasFields() {
+        return sourceFilterAliasFields;
+    }
+
+    public static class Builder {
+
+        private final List<String> caseTypeIds = new ArrayList<>();
+        private JsonNode searchRequestJsonNode;
+        private boolean multiCaseTypeSearch;
+        private final List<String> sourceFilterAliasFields = new ArrayList<>();
+
+        public Builder withCaseTypes(List<String> caseTypeIds) {
+            if (caseTypeIds != null) {
```

Comment:
is this null check required?

Reply:
addAll expects a NotNull collection, so have to check if null